### PR TITLE
ci(build): add gh action pipelines

### PR DESCRIPTION
**Pull request by mmpetarpeshev**
We have to sync on the github events and workflow. 
Also what should be value for the FAUCET_ACCOUNT_PRIV_KEY for each environment.

Will create and gitops folders for the app.

### CODE REVIEW

**Comment by marc0olo** [comment]
> Also what should be value for the FAUCET_ACCOUNT_PRIV_KEY for each environment.

for the faucet we only want and need 1 main environment. no need for staging or PR based deployments. at least right now. we only need production and we should use the account that is currently used for https://faucet.aepps.com (which should be replaced by this application here ;))

**Comment by dincho** [comment]
@marc0olo there is structure to be followed in the apps cluster, each chart MUST exists on dev and stg, eventually on prd.
@mmpetarpeshev the `FAUCET_ACCOUNT_PRIV_KEY` should be the same for all environments

**Comment by marc0olo** [comment]
> @marc0olo there is structure to be followed in the apps cluster, each chart MUST exists on dev and stg, eventually on prd.

ohh, that is good to know! I think @thepiwo had a similar discussion few weeks ago 😅

**Comment by thepiwo** [comment]
whatever it takes to fullfill the goal :shrug:

**Comment by marc0olo** [comment]
so the most important thing to me is still to get this somehow deployed in prd and have a reliable reflection of master -> prd

I am still blind here right now. in case we have to have dev and stg environment for reasons I suggest to secure those domains as we shouldn't expose the faucet aepp 3 times with same key IMO. everybody would be able to use the same faucet on 3 different URLs then. but we can also secure the domains afterwards. I just wanna proceed here somehow.

I feel like we over-complicate this meanwhile. in the end we just need the master branch here to be deployed in prd and to be accessible via https://faucet.aepps.com

**Comment by mmpetarpeshev** [comment]
Nothing complicated , just decide when you want this to be deployed on prod. 
It can be done by the requirements team requirements . 
If you deploy every commit ,is that safe no tests, etc...
Do you want to have release process for this application or no ? 
I can configure the pipelines as is needed , but someone should know what is that.

**Comment by marc0olo** [comment]
> I can configure the pipelines as is needed , but someone should know what are the requirements.

ok perfect. IMO we do not need dev or staging environment here. what would be very helpful is a PR based deployment for PRs against master branch so that we can review before moving into production.

this is only maintenance right now with no active development.

**Comment by marc0olo** [comment]
> Cool, lets we sync tomorrow and I will create the pipelines as we need them.

can we do it like proposed above in my last comment?

File: .github/workflows/staging-pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+env:
+  PR_NUMBER: ${{ github.event.pull_request.number }}
+  ENV: "stg"
+  APP: "aepp-faucet-nodejs"
+
+jobs:
+  main:
+    runs-on: ubuntu-latest    
+    name: Staging Aepp Faucet Nodejs Pipeline Pipeline
```
Comment (marc0olo) [inline .github/workflows/staging-pipeline.yml:16]:
```suggestion
    name: Staging Aepp Faucet Nodejs Pipeline
```

File: .github/workflows/prod-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      - master
+      
+env:
+  ENV: "prd"
+  APP: "aepp-faucet-nodejs"
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    name: Production Aepp Faucet Nodejs Pipeline Pipeline
```
Comment (marc0olo) [inline .github/workflows/prod-pipeline.yml:15]:
```suggestion
    name: Production Aepp Faucet Nodejs Pipeline
```

File: .github/workflows/prod-pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,108 @@
+name: Production Aepp Faucet Nodejs Pipeline Pipeline
```
Comment (marc0olo) [inline .github/workflows/prod-pipeline.yml:1]:
```suggestion
name: Production Aepp Faucet Nodejs Pipeline
```

File: .github/workflows/staging-pipeline.yml [hunk 4]
```diff
@@ -0,0 +1,132 @@
+name: Staging Aepp Faucet Nodejs Pipeline Pipeline
```
Comment (marc0olo) [inline .github/workflows/staging-pipeline.yml:1]:
```suggestion
name: Staging Aepp Faucet Nodejs Pipeline
```

**Comment by marc0olo** [comment]
other than the typo stuff I am not sure if I can 100% judge that.

as long as following conditions are met I am fine with merging it:
- on each PR against master a PR specific deployment is triggered
- on each merge/push into master a deployment to prod is triggered

**Comment by mmpetarpeshev** [comment]
> other than the typo stuff I am not sure if I can 100% judge that.
> 
> as long as following conditions are met I am fine with merging it:
> 
> * on each PR against master a PR specific deployment is triggered
> * on each merge/push into master a deployment to prod is triggered

Correct , thats what pipelines should do.Will fix the typo and let me know ,if something is not as expected.Its tested and should work as expected.

File: .github/workflows/prod-pipeline.yml (hunk 3 above)
Comment (mmpetarpeshev) [reply .github/workflows/prod-pipeline.yml]: Fixed

**Review by marc0olo — APPROVED** [review]
let's see how it goes ;-)

**Pull request merged**